feat: add attendee related functions

abhishek-gkmit opened the pull request:
add `AddAttendee`, `EditAttendee`, `AttendeeFormHelper` and `AttendeeList` components add type declarations
add styles for above components
add routes for `add` and `edit` attendee

abhishek-gkmit committed:
feat: add attendee related functions
add `AddAttendee`, `EditAttendee`, `AttendeeFormHelper` and `AttendeeList` components
add type declarations
add styles for above components
add routes for `add` and `edit` attendee

adiTrivsme commented:
<img width="867" alt="Screenshot 2024-10-04 at 4 22 37 PM" src="https://github.com/user-attachments/assets/a57982f7-900a-4860-ab91-82976ea6883e">

if you want mention multiple changes use a bullet point list for clarity

adiTrivsme commented on src/App.tsx:
```diff
@@ -1,8 +1,10 @@
 import { AddEvent } from "@components/AddEvent/AddEvent";
 import { EditEvent } from "@components/EditEvent/EditEvent";
-import { Dashboard } from "@components/Dashboard";
+import { Dashboard } from "@src/components/Dashboard/Dashboard";
```
why not using @component directly??

and why there's `Dashboard/Dashboard`??

adiTrivsme commented on src/App.tsx:
```diff
@@ -1,8 +1,10 @@
+import { Dashboard } from "@src/components/Dashboard/Dashboard";
 import { Layout } from "@components/Layout";
+import { AddAttendee } from "@components/AddAttendee/AddAttendee";
+import { EditAttendee } from "@components/EditAttendee/EditAttendee";
```
same here, simple `index.tsx` file can be used

adiTrivsme commented on src/components/AddAttendee/AddAttendee.tsx:
```diff
@@ -0,0 +1,20 @@
+    <AttendeeFormHelper
+      initFormData={{ name: "", email: "" }}
+      isEdit={false}
+      eventId={+eventId}
```
add comment for the use of `+`

adiTrivsme commented on src/components/AttendeeFormHelper/AttendeeFormHelper.tsx:
```diff
@@ -0,0 +1,78 @@
+  const navigate = useNavigate();
+
+  const event = useMemo(
+    () => events.find(({ id }) => id === eventId),
```
the dependency array is not complete you are missing something

adiTrivsme commented on src/components/EditAttendee/EditAttendee.tsx:
```diff
@@ -0,0 +1,32 @@
+export function EditAttendee() {
+  const navigate = useNavigate();
+  const { eventId, attendeeId } = useParams();
+  console.log(`eventId: ${eventId}, attendeeId: ${attendeeId}`);
```
remove console

adiTrivsme commented on src/components/EditAttendee/EditAttendee.tsx:
```diff
@@ -0,0 +1,32 @@
+    return;
+  }
+
+  const initFormData = events
```
do not use directly add in useMemo

abhishek-gkmit committed:
fix: alias imports
- rename all the component files to `index.tsx` in their respecitve directories
- add direct alias imports where it was being us indirectly

abhishek-gkmit committed:
fix: add comment for using `+eventId`

abhishek-gkmit committed:
fix: add remaining dependency in `useMemo`

abhishek-gkmit committed:
fix: remove unnecessary `console.log`

abhishek-gkmit committed:
fix: add exception for `event` in `AttendeeList` and rearrange imports
fix style imports

abhishek-gkmit committed:
fix: remove `Dashboard` and it's routes

abhishek-gkmit committed:
fix: add `*` for required input field indication
